openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+from collections import Iterable
+from numbers import Real, Integral
+import copy
+import sys
+
+import numpy as np
+
+import openmc.checkvalue as cv
+
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class EnergyGroups(object):
+    """An energy groups structure used for multi-group cross-sections.
+
+    Parameters
+    ----------
+    group_edges : ndarray
```

Comment:
Would it? The `group_edges` is cast to an `ndarray` in the `group_edges` property setter.